src/diff/index.js

```diff
@@ -3,6 +3,10 @@ import { isDate, isEmpty, isObject, properObject } from '../utils';
 const diff = (lhs, rhs) => {
   if (lhs === rhs) return {}; // equal return no diff
 
+  if (typeof lhs === "number" && typeof rhs === "number") {
+    if (isNaN(lhs) && isNaN(rhs)) return {}; // NaN is equal
+  }
```

Comment:
Could use [`Number.isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) to check both `Number` type and `NaN` value simultaneously.

```diff
-   if (typeof lhs === "number" && typeof rhs === "number") {
-     if (isNaN(lhs) && isNaN(rhs)) return {}; // NaN is equal
-  }
+   if (Number.isNan(lhs) && Number.isNan(rhs)) return {}; // NaN is equal
```

Reply:
@anko you meant to cap the last 'n' --> `Number.isNaN`